ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/ontologies/views/OntologiesViewImpl.java

```diff
@@ -0,0 +1,351 @@
+        Tree.TreeNode<OntologyHierarchy> node = tree.findNode(el);
+        if (node != null) {
+            return node.getModel();
+        }
+        return null;
+    }
+
+    @Override
+    public OntologyHierarchy getSelectedHierarchy() {
+        return tree.getSelectionModel().getSelectedItem();
+    }
+
+    @Override
+    public void clearStore() {
+        treeStore.clear();
+    }
+
+    @Override
+    public void addToStore(List<OntologyHierarchy> children) {
+        treeStore.add(children);
+    }
+
+    @Override
+    public void addToStore(OntologyHierarchy parent, List<OntologyHierarchy> children) {
+        treeStore.add(parent, children);
+    }
+
+    @UiFactory
+    SimpleComboBox<Ontology> createComboBox() {
+        final SimpleComboBox<Ontology> ontologySimpleComboBox = new SimpleComboBox<Ontology>(new LabelProvider<Ontology>() {
+            @Override
+            public String getLabel(Ontology item) {
+                if (item.isActive()){
+                    activeOntology = item;
+                    return item.getVersion() + "  <--- *(DE)*";
+                }
+                return item.getVersion();
+            }
+        });
+
+        ontologySimpleComboBox.setEditable(false);
+        ontologySimpleComboBox.setEmptyText(appearance.selectOntologyVersion());
+        ontologySimpleComboBox.setTriggerAction(ComboBoxCell.TriggerAction.ALL);
+        ontologySimpleComboBox.setWidth(500);
+        ontologySimpleComboBox.addSelectionHandler(new SelectionHandler<Ontology>() {
+            @Override
+            public void onSelection(SelectionEvent<Ontology> event) {
+                selectedOntology = event.getSelectedItem();
+                updateButtonStatus();
+                if (selectedOntology != null) {
+                    fireEvent(new SelectOntologyVersionEvent(selectedOntology));
+                }
+
+            }
+        });
+        return ontologySimpleComboBox;
+    }
+
+    void updateButtonStatus() {
+        publishButton.setEnabled(selectedOntology != null && selectedOntology != activeOntology);
+        saveHierarchy.setEnabled(selectedOntology != null);
+        categorize.setEnabled(selectedOntology != null && targetApp != null);
+    }
+    @UiHandler("viewVersions")
+    void viewVersionsClicked(SelectEvent event) {
+        fireEvent(new ViewOntologyVersionEvent());
+    }
+
+    @UiHandler("publishButton")
+    void publishButtonClicked(SelectEvent event) {
+        Ontology editedOntology = ontologyDropDown.getCurrentValue();
+        if (editedOntology != null){
+            new PublishOntologyDialog(appearance, editedOntology, activeOntology, this);
+        }
+
+    }
+
+    @UiHandler("addButton")
+    void addButtonClicked(SelectEvent event) {
+        new EdamUploadDialog(UriUtils.fromTrustedString(clientConstants.ontologyUploadServlet())).show();
+    }
+
+    @UiHandler("saveHierarchy")
+    void saveHierarchyClicked(SelectEvent event) {
+        treeStore.clear();
```

Comment:
The treeStore should probably not be cleared until it's ready to be reloaded (when a service call successfully returns with a new hierarchy listing).
Clearing the store here means that the admin can click the "Save Hierarchies" button, then change their mind and click "Cancel" on the dialog, and the previously loaded hierarchies are cleared but not replaced with a new listing.